tests/accelerators/test_gpu.py

```diff
@@ -72,3 +72,15 @@ def test_gpu_availability():
 def test_warning_if_gpus_not_used():
     with pytest.warns(UserWarning, match="GPU available but not used. Set `accelerator` and `devices`"):
         Trainer()
+
+
+@RunIf(min_gpus=1)
+def test_gpu_teardown():
+    acc = GPUAccelerator()
+    tensor = torch.zeros(100).cuda()
+    del tensor
```

Comment:
are we counting on garbage collection being done at the appropriate time here?
It might be safer to add a `gc.collect()` call?